src/compiler/transformers/map-imports-to-path-aliases.ts

```diff
@@ -0,0 +1,60 @@
+import { normalize, relative } from 'path';
+import ts from 'typescript';
+import type * as d from '../../declarations';
+
+/**
+ * This method is responsible for replacing user-defined import path aliases ({@link https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping})
+ * with generated relative import paths during the transformation step of the TS compilation process.
+ * This action is taken to prevent issues with import paths not being transpiled at build time resulting in
+ * unknown imports in output code for some output targets (`dist-collection` for instance). Output targets that do not run through a bundler
+ * are unable to resolve imports using the aliased path names and TS intentionally does not replace resolved paths as a part of
+ * their compiler ({@link https://github.com/microsoft/TypeScript/issues/10866})
+ *
+ * @param config The Stencil configuration object.
+ * @returns A factory for creating a {@link ts.Transformer}.
+ */
+export const mapImportsToPathAliases = (config: d.Config): ts.TransformerFactory<ts.SourceFile> => {
```

Comment:
Since this is only called by `src/compiler/output-targets/dist-collection/index.ts#outputCollection`, and `outputCollection` passes a config of type `ValidatedConfig` to this fn, can we update this function signature accordingly?
```suggestion
export const mapImportsToPathAliases = (config: d.ValidatedConfig): ts.TransformerFactory<ts.SourceFile> => {
```

Reply:
Ah yep, good catch. Couldn't use the ValidatedConfig when this was running as a part of the main transpilation process